homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -1182,6 +1208,50 @@ async fn handle_swarm_event<DB: Database>(
                   "outgoing connection error"
             );
 
+            // Redial peer if in configured peers
+            if let Some(peer_id) = peer_id {
+                if event_handler.node_addresses.iter().any(|multiaddr| {
+                    if let Some(id) = multiaddr.peer_id() {
+                        id == peer_id
+                    } else {
+                        // TODO: We may want to check the multiadress without relying on
+                        // the peer ID. This would give more flexibility when configuring nodes.
+                        warn!(
+                            subject = "libp2p.conn.closed",
```

Comment:
We should update the subject here to `libp2p.outgoing.err`.

Also, maybe worth a helper function now that we are using this conditional check twice?

Reply:
Thanks! I updated the subject, but haven't extracted to a helper function yet, because of the logging that happens inside of the conditional check that needs to be different at both call sites. If you think it's worthwhile, we can just pass the subject as a parameter though.

Reply:
Yeah, that does make it a bit awkward because it's not just a conditional check. I'm alright leaving it as is. 👌